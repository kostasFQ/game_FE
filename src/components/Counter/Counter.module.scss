.counter {
  border: solid red 1px;
  z-index: -1;
  position: absolute;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 55%;
  width: 100vw;
  
  &__digit {
    height: 90%;
    border: solid green 1px;
    width: 100vw;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 35em;
  }
}