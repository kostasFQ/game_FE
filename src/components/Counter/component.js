import React from 'react';
import styles from './Counter.module.scss';
import PropTypes from 'prop-types';

const Counter = ({ clicks }) => (
  <div className={styles.counter}>
    <div>your points</div>
    <div className={styles.counter__digit}>{ clicks }</div>
    <div>times</div>
  </div>
)

Counter.propTypes = {
  clicks: PropTypes.number.isRequired
}

export default Counter;