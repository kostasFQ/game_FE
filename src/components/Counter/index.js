import Counter from './CounterComponent';

export default Counter;