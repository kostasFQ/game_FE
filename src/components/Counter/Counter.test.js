import React from 'react';
import Counter from './component';
import renderer from 'react-test-renderer';

test('Counter should be render with 5 clicks', () => {
  const instance = renderer.create(
    <Counter title='Click Here!' clicks={ 5 } />
  );
  let component = instance.toJSON();
  expect(component).toMatchSnapshot();
})

test('Counter should be render with 10 clicks', () => {
  const instance = renderer.create(
    <Counter title='Click Here!' clicks={ 10 } />
  );
  let component = instance.toJSON();
  expect(component).toMatchSnapshot();
})