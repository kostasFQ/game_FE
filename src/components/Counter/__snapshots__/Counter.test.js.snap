// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should be render with 5 clicks 1`] = `
<div
  className="counter"
>
  <div>
    YOU CLICK
  </div>
  <div
    className="counter__digit"
  >
    5
  </div>
  <div>
    times
  </div>
</div>
`;

exports[`Counter should be render with 10 clicks 1`] = `
<div
  className="counter"
>
  <div>
    YOU CLICK
  </div>
  <div
    className="counter__digit"
  >
    10
  </div>
  <div>
    times
  </div>
</div>
`;
